src/Microsoft.AspNet.Mvc.TagHelpers/JavaScriptUtility.cs

```diff
@@ -68,6 +68,32 @@ internal static string PrepareFormatString(string input)
         }
 
         /// <summary>
+        /// Encodes a .NET string array for safe use as a JavaScript array literal, including inline in an HTML file.
+        /// </summary>
+        internal static string JavaScriptArrayEncode(IEnumerable<string> values)
+        {
+            var result = new StringBuilder();
+            result.Append("[");
+
+            var firstAdded = false;
+
+            foreach (var value in values)
+            {
+                if (firstAdded)
+                {
+                    result.Append(",");
+                }
+
+                result.AppendFormat("\"{0}\"", JavaScriptStringEncode(value));
```

Comment:
That would require me to change how the embedded JS is loaded from the assembly too, as currently the params are passed in via an `Append.Format` call.